lib/queries/upsertOne.js

```diff
@@ -23,22 +25,37 @@ export default ({
 
         const setQuery = writableCols
             .filter(column => typeof parameters[column] !== 'undefined')
-            .map((column) => `${column} = $${column}`);
+            .map(column => `${column} = $${column}`);
+
+        const permanentFiltersKeys = Object.keys(permanentFilters);
+        let where = '';
+        let permanentParameters = {};
+        if (permanentFiltersKeys.length) {
+            permanentParameters = sanitizeIdentifier(permanentFiltersKeys, permanentFilters);
+            where = ` ${whereQuery(permanentParameters, permanentFiltersKeys)}`;
+        }
 
         const sql = (
 `INSERT INTO ${table} (${keys.join(', ')})
 VALUES (${values})
 ON CONFLICT (${primaryKey.join(', ')})
 ${
 setQuery.length > 0 ? (
-    `DO UPDATE SET ${setQuery.join(', ')}`
+    `DO UPDATE SET ${setQuery.join(', ')}${where}`
```

Comment:
```js
${where ? ` ${where}` : ''}
```

Reply:
ternary operator is applied  [here](https://github.com/marmelab/coPostgresQuery/pull/48/commits/10f21729a60ca63617cc0fc21390e8c5cc7c91a2#diff-a6df500cd865db6e20b013575ab58a7fR31) (line 31)